scripts/write-config.sh

```diff
@@ -157,90 +108,80 @@ if [ -e "${outputdir}/target.txt" ]; then
     fi
 fi
 
-# Run vault either using a docker container or using the installed vault
-function dovault {
-    local dovaultpath=$1
-    local dofilename=$2
-    case $vaultenv in
-        docker)
-            docker run --rm -e VAULT_TOKEN="${vaulttoken}" broadinstitute/dsde-toolbox:dev \
-                   vault read -format=json "${dovaultpath}" > "${dofilename}"
-            ;;
-
-        local)
-            VAULT_TOKEN="${vaulttoken}" VAULT_ADDR="https://clotho.broadinstitute.org:8200" \
-                   vault read -format=json "${dovaultpath}" > "${dofilename}"
-            ;;
-    esac
+# Fetch a secret from GSM
+function dogsm {
+    local dosecretproject=$1
+    local dovaultpath=$2
+    local dofilename=$3
+    gcloud secrets versions access latest --project "${dosecretproject}" --secret "${dovaultpath}" > "${dofilename}"
 }
 
-# Read a vault path into an output file, decoding from base64
+# Read a GSM secret into an output file, decoding from base64
 # To detect missing tokens, we need to capture the docker result before
 # doing the rest of the pipeline.
-function vaultgetb64 {
-    vaultpath=$1
-    filename=$2
+function gsmgetb64 {
+    secretproject=$1
+    vaultpath=$2
```

Comment:
What is `vaultpath`?  It's not still vault is it?

Reply:
Ah, missed that. It should be `secretname` now, I'll rename it